### ISSUE TITLE
[Bug]: ResizePayload panic with case: tpch dataset, `select * from lineitem`

### ISSUE THREAD

**Issue by xzxiong** [matrixorigin/matrixone#5712]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): d63fef2
- Hardware parameters: mo-dev1
- OS type: linux
- Others:
`./mo-service -launch ./etc/launch-tae-logservice/launch.toml &>mo-service.log &`
```

### Actual Behavior

- prepare
1G tpch dataset

- prepare table
```
Mysql> create database tpch;
Mysql> use tpch;
Mysql> CREATE TABLE LINEITEM(
L_ORDERKEY    BIGINT NOT NULL,
L_PARTKEY     INTEGER NOT NULL,
L_SUPPKEY     INTEGER NOT NULL,
L_LINENUMBER  INTEGER NOT NULL,
L_QUANTITY    DECIMAL(15,2) NOT NULL,
L_EXTENDEDPRICE  DECIMAL(15,2) NOT NULL,
L_DISCOUNT    DECIMAL(15,2) NOT NULL,
L_TAX         DECIMAL(15,2) NOT NULL,
L_RETURNFLAG  VARCHAR(1) NOT NULL,
L_LINESTATUS  VARCHAR(1) NOT NULL,
L_SHIPDATE    DATE NOT NULL,
L_COMMITDATE  DATE NOT NULL,
L_RECEIPTDATE DATE NOT NULL,
L_SHIPINSTRUCT CHAR(25) NOT NULL,
L_SHIPMODE     CHAR(10) NOT NULL,
L_COMMENT      VARCHAR(44) NOT NULL
);
```

- exec load
```
MySQL [system]> use tpch ;
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
MySQL [tpch]> load data infile '/data/go/src/github.com/matrixorigin/1G_data/lineitem.tbl' into table LINEITEM FIELDS TERMINATED BY '|' OPTIONALLY ENCLOSED BY '"' LINES TERMINATED BY '\n';
Query OK, 6001215 rows affected (15.99 sec)

MySQL [tpch]> select now(); select count(*) from lineitem; select now();
+----------------------------+
| now()                      |
+----------------------------+
| 2022-10-18 16:02:16.928294 |
+----------------------------+
1 row in set (0.00 sec)

ERROR 2013 (HY000): Lost connection to MySQL server during query
ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
ERROR 2003 (HY000): Can't connect to MySQL server on '127.1' (111)
ERROR: Can't connect to the server

unknown [tpch]> \q
```

- panic error
```
2022/10/18 16:02:34.705563 +0800 DEBUG LogService morpc/server.go:178 received request {"uuid": "7c4dccb4-4d3c-41f8-b482-5251dc7a41bf", "name": "logservice-rpc", "sequence": 3, "client": "127.0.0.1:55806", "request-id": 3, "request": ""}
2022/10/18 16:02:34.712158 +0800 DEBUG LogService v2@v2.0.3-0.20220926012906-f97d6dabbb6f/application.go:328 session read message {"uuid": "7c4dccb4-4d3c-41f8-b482-5251dc7a41bf", "name": "logservice-rpc", "listen-address": "127.0.0.1:32001", "session-id": 90, "addr": "127.0.0.1:55892", "seqence": 3}
2022/10/18 16:02:34.712252 +0800 DEBUG LogService morpc/server.go:178 received request {"uuid": "7c4dccb4-4d3c-41f8-b482-5251dc7a41bf", "name": "logservice-rpc", "sequence": 3, "client": "127.0.0.1:55892", "request-id": 3, "request": ""}
2022/10/18 16:02:34.716664 +0800 DEBUG LogService v2@v2.0.3-0.20220926012906-f97d6dabbb6f/application.go:328 session read message {"uuid": "7c4dccb4-4d3c-41f8-b482-5251dc7a41bf", "name": "logservice-rpc", "listen-address": "127.0.0.1:32001", "session-id": 29, "addr": "127.0.0.1:55766", "seqence": 3}
2022/10/18 16:02:34.716753 +0800 DEBUG LogService morpc/server.go:178 received request {"uuid": "7c4dccb4-4d3c-41f8-b482-5251dc7a41bf", "name": "logservice-rpc", "sequence": 3, "client": "127.0.0.1:55766", "request-id": 3, "request": ""}
2022/10/18 16:02:34.717526 +0800 DEBUG LogService v2@v2.0.3-0.20220926012906-f97d6dabbb6f/application.go:328 session read message {"uuid": "7c4dccb4-4d3c-41f8-b482-5251dc7a41bf", "name": "logservice-rpc", "listen-address": "127.0.0.1:32001", "session-id": 21, "addr": "127.0.0.1:55750", "seqence": 3}
2022/10/18 16:02:34.717582 +0800 DEBUG LogService morpc/server.go:178 received request {"uuid": "7c4dccb4-4d3c-41f8-b482-5251dc7a41bf", "name": "logservice-rpc", "sequence": 3, "client": "127.0.0.1:55750", "request-id": 3, "request": ""}
panic: runtime error: slice bounds out of range [:2198996] with capacity 2198994

goroutine 167403 [running]:
github.com/matrixorigin/matrixone/pkg/pb/logservice.(*LogRecord).ResizePayload(...)
        /data/go/src/github.com/matrixorigin/matrixone/pkg/pb/logservice/logservice.go:35
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/driver/logservicedriver.(*driverAppender).append(0xc00dbae340, 0x4e01?, 0x3b9aca00)
        /data/go/src/github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/driver/logservicedriver/appender.go:54 +0x3ae
created by github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/driver/logservicedriver.(*LogServiceDriver).onAppendQueue
        /data/go/src/github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/driver/logservicedriver/append.go:66 +0xf2
```

**Comment by jiangxinmeng1** [comment]
Before ResizePayload, appender has checked the size of the record. 

<img width="246" alt="3" src="https://user-images.githubusercontent.com/51114574/199496080-1f3fc8e6-b4ba-4244-8a99-7ff615a25a2a.PNG">

The size in logservice.LogRecord.ResizePayload() doesn't match the length of the payload.

<img width="193" alt="1" src="https://user-images.githubusercontent.com/51114574/199496055-0dd7c552-c386-44da-8402-c39a51594c3e.PNG">

logservice.LogRecord.Size() also doesn't match.

<img width="743" alt="4" src="https://user-images.githubusercontent.com/51114574/199496248-ddd6d2db-dc34-4e94-890f-f5c01a9b7a7b.PNG">

**Comment by volgariver6** [comment]
Has been fixed by commit 9a27c2b.
